cmd/subctl/join.go

```diff
@@ -260,6 +260,10 @@ func determineClusterID(status reporter.Interface) {
 	if joinFlags.ClusterID == "" {
 		joinFlags.ClusterID, err = restConfigProducer.GetClusterID()
 		exit.OnError(status.Error(err, "Error determining cluster ID of the target cluster"))
+
+		if err = cluster.IsValidID(joinFlags.ClusterID); err != nil {
```

Comment:
Some print line stating that the cluster ID is internally sanitized avoids confusion when the user puts in an invalid ID and then all the logs/progress mentions the sanitized ID. I would want to know if the code changes something that I had input.

Reply:
> Some print line stating that the cluster ID is internally sanitized avoids confusion when the user puts in an invalid ID and then all the logs/progress mentions the sanitized ID. I would want to know if the code changes something that I had input.

If user inputs an invalid ID it will throw error and fail instead of using sanitized version. We sanitize only in scenarios where clusterID is extracted from kubeconfig.